sql/src/main/antlr/OpenSearchSQLParser.g4

```diff
@@ -320,6 +321,10 @@ highlightFunction
     : HIGHLIGHT LR_BRACKET relevanceField (COMMA highlightArg)* RR_BRACKET
     ;
 
+matchQueryAltSyntaxFunction
+    : field=relevanceField EQUAL_SYMBOL MATCH_QUERY LR_BRACKET query=relevanceQuery RR_BRACKET
```

Comment:
replace `MATCH_QUERY` with something like `equalFieldRelevanceFunctionName` and make this a list that includes `MATCH_QUER` so we can add more items.